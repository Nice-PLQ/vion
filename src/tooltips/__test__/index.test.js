import { mount } from '@vue/test-utils';
import Tooltips from '../tooltips.vue';
import { next } from '../../../test/index';

describe('Tooltips', () => {
  test('渲染', async () => {
    const wrapper = mount(Tooltips, {
      slots: {
        default: '<button>按钮</button>',
      },
      props: {
        visible: false,
        text: '气泡提示的内容',
        placement: 'top-center'
      }
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  test('显示与隐藏', async () => {
    const wrapper = mount(Tooltips, {
      slots: {
        default: '<button>按钮</button>',
      },
      props: {
        visible: false,
        text: '气泡提示的内容',
        placement: 'top-center'
      }
    });
    expect(wrapper.find('.v-tooltips__inner').isVisible()).toBe(false);
    wrapper.setProps({
      visible: true,
    });
    await next();
    expect(wrapper.find('.v-tooltips__inner').isVisible()).toBe(true);
  });

  test('改变位置', async () => {
    const wrapper = mount(Tooltips, {
      slots: {
        default: '<button>按钮</button>',
      },
      props: {
        visible: true,
        text: '气泡提示的内容',
        placement: 'top-left'
      }
    });
    const wrapper1 = mount(Tooltips, {
      slots: {
        default: '<button>按钮</button>',
      },
      props: {
        visible: true,
        text: '气泡提示的内容',
        placement: 'top-right'
      }
    });
    const wrapper2 = mount(Tooltips, {
      slots: {
        default: '<button>按钮</button>',
      },
      props: {
        visible: true,
        text: '气泡提示的内容',
        placement: 'bottom-center'
      }
    });
    const wrapper3 = mount(Tooltips, {
      slots: {
        default: '<button>按钮</button>',
      },
      props: {
        visible: true,
        text: '气泡提示的内容',
        placement: 'bottom-left'
      }
    });
    const wrapper4 = mount(Tooltips, {
      slots: {
        default: '<button>按钮</button>',
      },
      props: {
        visible: true,
        text: '气泡提示的内容',
        placement: 'bottom-right'
      }
    });
    expect(wrapper.element).toMatchSnapshot();
    expect(wrapper1.element).toMatchSnapshot();
    expect(wrapper2.element).toMatchSnapshot();
    expect(wrapper3.element).toMatchSnapshot();
    expect(wrapper4.element).toMatchSnapshot();
  });
});
