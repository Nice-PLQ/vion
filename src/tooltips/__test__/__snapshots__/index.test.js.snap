// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltips 改变位置 1`] = `
<div
  class="v-tooltips v-tooltips__bottom-left"
>
  <transition-stub
    appear="false"
    css="true"
    name="tooltips-top"
    persisted="true"
  >
    <div
      class="v-tooltips__inner"
      style="left: 0px; top: 0px;"
    >
      
      <div
        class="v-tooltips__txt-tips"
      >
        气泡提示的内容
      </div>
      
    </div>
  </transition-stub>
  <span>
    
    <button>
      按钮
    </button>
    
  </span>
</div>
`;

exports[`Tooltips 改变位置 2`] = `
<div
  class="v-tooltips v-tooltips__bottom-right"
>
  <transition-stub
    appear="false"
    css="true"
    name="tooltips-top"
    persisted="true"
  >
    <div
      class="v-tooltips__inner"
      style="left: 0px; top: 0px;"
    >
      
      <div
        class="v-tooltips__txt-tips"
      >
        气泡提示的内容
      </div>
      
    </div>
  </transition-stub>
  <span>
    
    <button>
      按钮
    </button>
    
  </span>
</div>
`;

exports[`Tooltips 改变位置 3`] = `
<div
  class="v-tooltips v-tooltips__top-center"
>
  <transition-stub
    appear="false"
    css="true"
    name="tooltips-bottom"
    persisted="true"
  >
    <div
      class="v-tooltips__inner"
      style="left: 0px; top: 0px;"
    >
      
      <div
        class="v-tooltips__txt-tips"
      >
        气泡提示的内容
      </div>
      
    </div>
  </transition-stub>
  <span>
    
    <button>
      按钮
    </button>
    
  </span>
</div>
`;

exports[`Tooltips 改变位置 4`] = `
<div
  class="v-tooltips v-tooltips__top-left"
>
  <transition-stub
    appear="false"
    css="true"
    name="tooltips-bottom"
    persisted="true"
  >
    <div
      class="v-tooltips__inner"
      style="left: 0px; top: 0px;"
    >
      
      <div
        class="v-tooltips__txt-tips"
      >
        气泡提示的内容
      </div>
      
    </div>
  </transition-stub>
  <span>
    
    <button>
      按钮
    </button>
    
  </span>
</div>
`;

exports[`Tooltips 改变位置 5`] = `
<div
  class="v-tooltips v-tooltips__top-right"
>
  <transition-stub
    appear="false"
    css="true"
    name="tooltips-bottom"
    persisted="true"
  >
    <div
      class="v-tooltips__inner"
      style="left: 0px; top: 0px;"
    >
      
      <div
        class="v-tooltips__txt-tips"
      >
        气泡提示的内容
      </div>
      
    </div>
  </transition-stub>
  <span>
    
    <button>
      按钮
    </button>
    
  </span>
</div>
`;

exports[`Tooltips 渲染 1`] = `
<div
  class="v-tooltips v-tooltips__bottom-center"
>
  <transition-stub
    appear="false"
    css="true"
    name="tooltips-top"
    persisted="true"
  >
    <div
      class="v-tooltips__inner"
      style="left: 0px; top: 0px; display: none;"
    >
      
      <div
        class="v-tooltips__txt-tips"
      >
        气泡提示的内容
      </div>
      
    </div>
  </transition-stub>
  <span>
    
    <button>
      按钮
    </button>
    
  </span>
</div>
`;
